crates/autopilot/src/run_loop.rs

```diff
@@ -765,6 +835,10 @@ struct Metrics {
     /// solved and before sending a `settle` request.
     auction_postprocessing_time: prometheus::Histogram,
 
+    /// Tracks the time spent in pre-processing before sending a `solve`
+    /// request.
+    auction_preprocessing_time: prometheus::Histogram,
```

Comment:
If we avoid cloning the auction only a simple `auction.orders.retain(|o| !in_flight.contains(&o))` should happen in this step which is so fast that measuring it doesn't make sense.